pkg/proxy/proxy.go

```diff
@@ -349,11 +349,30 @@ func BootProxy(logger *zap.Logger, opt Option, appCmd, appContainer string, pid
 	if opt.Port == 0 {
 		opt.Port = 16789
 	}
+	maxAttempts := 1000
+	attemptsDone:=0
 
+
+    if !isPortAvailable(opt.Port) {
+		for i := 1024; i <= 65535 && attemptsDone < maxAttempts; i++ {
+			if isPortAvailable(uint32(i)) {
+				opt.Port = uint32(i)
+				logger.Info("Port %v is already in use, switching to next available port");
+
+				break
+			}
+			attemptsDone++
+		}
+	}
+	
+	if maxAttempts <= attemptsDone{
+		logger.Error("Failed to find an available port to start proxy server")
+		return nil
+	}
 	//IPv4
 	localIp4, err := util.GetLocalIPv4()
 	if err != nil {
-		log.Fatalln(Emoji+"Failed to get the local Ip4 address", err)
+		log.Fatalln("Failed to get the local Ip4 address", err)
```

Comment:
Why is it removed from here? This is not zap.logger, It is just log. Here we should use emoji.